python/aie/compiler/aiecc/main.py

```diff
@@ -83,9 +86,18 @@ async def do_call(self, task, command, force=False):
   def do_run(self, command):
       if(self.opts.verbose):
           print(" ".join(command))
-      ret = run(command, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+      m = subprocess.PIPE
+      ret = subprocess.run(command, stdout=m, stderr=m, universal_newlines=True)
       return ret
 
+  def run_passes(self, pass_pipeline, mlir_module, outputfile=None):
+      if self.opts.verbose:
+        print("Running:", pass_pipeline)
+      PassManager.parse(pass_pipeline).run(mlir_module)
+      if outputfile:
+        with open(outputfile, 'w') as g:
+          g.write(str(mlir_module))
```

Comment:
@fifield 

My recommendation is to narrow the extent/scope for the `with Context() as ctx, Location.unknown()`. Currently you have nested contexts (one with in `main` and then another in `run_flow`). You could collapse them both to being just here; if instead of parsing the module in `main` and then holding onto it as a `ir.Module`, you just held the string and only constructed the `ir.Module` here (i.e., just before running the PassManager on it).

Reply:
Good point. I store and parse the module as a string in `run_flow` because I need to have a new context due to the multi-threading (at least I think that's what's going on, it didn't work with a single context).